sdcm/cluster.py

```diff
@@ -1703,10 +1705,32 @@ def get_ldap_config():
                 'ldap_bind_dn': f'cn=admin,{LDAP_BASE_OBJECT}',
                 'ldap_bind_passwd': LDAP_PASSWORD}
 
+    @staticmethod
+    def get_saslauthd_config():
+        if Setup.LDAP_ADDRESS is None:
+            return {}
+        ldap_server_ip = '127.0.0.1' if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION else Setup.LDAP_ADDRESS[0]
+        ldap_port = LDAP_SSH_TUNNEL_LOCAL_PORT if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION else \
+            Setup.LDAP_ADDRESS[1]
+        return {'ldap_servers': f'ldap://{ldap_server_ip}:{ldap_port}/',
+                'ldap_search_base': f'ou=Person,{LDAP_BASE_OBJECT}',
+                'ldap_bind_dn': f'cn=admin,{LDAP_BASE_OBJECT}',
+                'ldap_bind_pw': LDAP_PASSWORD}
+
     def create_ldap_users_on_scylla(self):
         self.run_cqlsh(f'CREATE ROLE \'{LDAP_ROLE}\' WITH SUPERUSER=true')
         for user in LDAP_USERS:
-            self.run_cqlsh(f'CREATE ROLE \'{user}\' WITH login=true AND password=\'{LDAP_PASSWORD}\'')
+            # Cannot create passwords with SaslauthdAuthenticator
+            self.run_cqlsh(f'CREATE ROLE \'{user}\' WITH login=true')
+        result = self.remoter.run("grep -o '^authenticator: .*' /etc/scylla/scylla.yaml")
```

Comment:
for `branch-2021.1` it is much better to use the `node.remote_scylla_yaml` call to edit `scylla.yaml` file

Reply:
The grep command isn't editing the yaml file.

Reply:
well, it can be used even to read a yaml file and see it as a dict object, then you can do something like:
```
with self.remote_scylla_yaml() as scylla_yaml:
    if 'authenticator' in scylla_yaml:
        if scylla_yaml['authenticator'] == 'com.scylladb.auth.SaslauthdAuthenticator':
......
```

Reply:
Thanks, just sent a cleanup for sct master: https://github.com/scylladb/scylla-cluster-tests/pull/3445